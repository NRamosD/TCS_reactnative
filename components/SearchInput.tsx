import { useState } from "react";
import { Button, Image, StyleSheet, Text, TextInput, TouchableOpacity, View } from "react-native";


export function SearchInput({searchText, setSearchText}:any) {
    // const [searchText, setSearchText] = useState('');
    return (
        <View style={styles.searchContainer}>
            <View>
                <TextInput
                    style={styles.inputDesign}
                    placeholder="Search..."
                    onChangeText={newText => setSearchText(newText)}
                    defaultValue={searchText}
                />
                {
                    searchText.length>0?
                    <TouchableOpacity  onPress={()=>setSearchText("")} style={{ position:"absolute", right:10, top:15}}>
                        <Image source={require("@/assets/images/close.png")}
                            style={{width:10, height:10}}
                        />
                    </TouchableOpacity>
                    :null
                }
            </View>
            {
                searchText.length>0?
                (
                    <>
                        <View style={styles.contResults}>
                            <Text style={{padding: 5}}>
                                Results for: {searchText}
                            </Text>
                            
                        </View>
                    
                    </>
                )
                :null
            }
        </View>
    );
}

const styles = StyleSheet.create({
    searchContainer:{
        height:80
    },
    inputDesign:{
        height:40,
        borderWidth:1,
        borderColor:"#e6e6e6",
        borderRadius:5,
        padding:10,
        paddingRight:30,
        color:"#02215f"
    },
    contResults:{
        flex: 1,
        flexDirection: 'row',
        borderBottomWidth:1,
        borderColor:"#e6e6e6",
        borderRadius:5,
        padding:5,
    },
});