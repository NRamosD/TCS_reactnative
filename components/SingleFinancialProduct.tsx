import { FinancialProduct } from "@/app/types/financialProduct";
import { useEffect, useState } from "react";
import { Image, ScrollView, StyleSheet, Text, View } from "react-native";



interface IProps{
    dataItem: FinancialProduct
}


export function SingleFinancialProduct({dataItem}:IProps) {
    
    console.log(dataItem)
  return (
    <View style={styles.contEachItem}>
        <View style={styles.contText}>
            <Text>
                {dataItem.name}
            </Text>
            <Text style={{fontSize:12}}>
                ID: {dataItem.id}
            </Text>
        </View>
        <View style={styles.contArrow}>
            <Image source={require('../assets/images/rightarrow.png')}
            style={{width: 10, height: 10, tintColor:"#d1d1d1"}}/>
        </View>
    </View>
  );
}

const styles = StyleSheet.create({
  contEachItem:{
    flex: 1,
    flexDirection: 'row',
    borderBottomWidth:1,
    borderColor:"#e6e6e6",
    borderRadius:5,
    padding:5,
  },
  contText:{width:"80%"},
  contArrow:{width:"20%", display:"flex", justifyContent:"center", alignItems:"flex-end", paddingRight:5},
});
