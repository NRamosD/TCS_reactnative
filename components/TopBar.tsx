import { StyleSheet, Text, View } from "react-native";


export function TopBar() {

  return (
    <View style={styles.contTopBar}>
        <View>
            <Text style={{textAlign:"center"}}>ðŸ’µ BANCO</Text>
        </View>
    </View>
  );
}

const styles = StyleSheet.create({
  contTopBar:{
    position:"absolute",
    top:25,
    width:"100%",
    height:40,
    alignContent:"center",
    textAlign:"center",
    backgroundColor:"#fff",
    display:"flex",
    justifyContent:"center",
    borderBottomWidth:1,
    borderColor:"#e6e6e6",
    zIndex:10
  }
});
