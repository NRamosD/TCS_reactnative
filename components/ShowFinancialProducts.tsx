import { useEffect, useState } from "react";
import { ScrollView, StyleSheet, Text, TouchableOpacity, View } from "react-native";
import { FlatList, GestureHandlerRootView } from "react-native-gesture-handler";
import { SingleFinancialProduct } from "./SingleFinancialProduct";
import { NavigationProp, ParamListBase, useNavigation } from "@react-navigation/native";
import { AUTHOR_ID, BASE_URL } from "@/constants/env";
import { FinancialProduct } from "@/app/types/financialProduct";


type Movie = {
    id: string;
    title: string;
    releaseYear: string;
  };


export function ShowFinancialProducts({searchText}:any) {
    const navigation: NavigationProp<ParamListBase> = useNavigation();


    const [isLoading, setLoading] = useState(true);
    const [data, setData] = useState<FinancialProduct[]>([]);
    const [filteredData, setFilteredData] = useState<FinancialProduct[]>([]);
    useEffect(()=>{
        getFinancialProduct()
    },[])

    

    useEffect(()=>{
      setTimeout(() => {
        const auxDataCont = [...data]
        const result = auxDataCont.filter(x=>x.name.toLowerCase().includes(searchText.toLowerCase()))
        setFilteredData(result)
      }, 2000);
    },[searchText])
  
    const getFinancialProduct = async () => {
      try {
        const response = await fetch(
        BASE_URL+"/bp/products",
        {
            method:"GET",
            headers: {   
              "authorId":`${AUTHOR_ID}`,
              "Access-Control-Allow-Origin":"*",
              "content-type":"application/json"
            }
        })
        const json = await response.json();
        console.log(json)
        setData(json);
      } catch (error) {
        console.error(error);
      } finally {
        setLoading(false);
      }
    };

    const navigateToDetail = (financialProduct: FinancialProduct) => {
        navigation.navigate('screens/DetailsScreen', { financialProduct });
    };

  return (
    <GestureHandlerRootView>
        <ScrollView style={styles.contScrollItems}>
            {isLoading ? (
            <Text>Loading...</Text>
        ) : (
          searchText.length>0?
          (
            filteredData.map((item)=> 
                <TouchableOpacity onPress={()=>navigateToDetail(item)}>
                    <SingleFinancialProduct dataItem={item}/>
                </TouchableOpacity>
            )
          )
          :
          (
            data.map((item)=> 
                <TouchableOpacity onPress={()=>navigateToDetail(item)}>
                    <SingleFinancialProduct dataItem={item}/>
                </TouchableOpacity>
            )
          )
            

            
        )}
        </ScrollView>
    </GestureHandlerRootView>
  );
}

const styles = StyleSheet.create({
  contScrollItems:{
    maxHeight:"70%",
    overflow:"scroll",
    padding:5,
    borderWidth:1,
    borderColor:"#e6e6e6"
  },
  contEachItem:{
    borderBottomWidth:1,
    borderColor:"#e6e6e6",
    borderRadius:5,
    padding:5,
    display:"flex"
  },
});



/*
<FlatList
            data={data}
            keyExtractor={({id}) => id}
            renderItem={({item}) => (
                <SingleFinancialProduct dataItem={item}/>
                // <View style={styles.contEachItem}>
                //     <View>
                //         <Text>
                //             Name
                //         </Text>
                //         <Text style={{fontSize:10}}>
                //             {item.title}, {item.releaseYear}
                //         </Text>
                //     </View>
                //     <View>
                //         <Text>
                //             ➡️
                //         </Text>
                //     </View>
                // </View>
            )}
            />


*/