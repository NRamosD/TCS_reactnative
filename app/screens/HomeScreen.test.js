// import { StatusBar } from "react-native";
import { render, fireEvent } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import { StatusBar } from "expo-status-bar";
import { TopBar } from '@/components/TopBar';

it('renders correctly StatusBar', () => {
    const tree = renderer.create(<StatusBar />).toJSON();
    expect(tree).toMatchSnapshot();
});
it('renders correctly Topbar', () => {
    const tree = renderer.create(<TopBar />).toJSON();
    expect(tree).toMatchSnapshot();
});


// import React from 'react';
// import { render, fireEvent } from '@testing-library/react-native';
// import HomeScreen from '../HomeScreen'; // Ajusta la ruta según tu estructura de carpetas
// import { NavigationContainer } from '@react-navigation/native';
// import renderer from 'react-test-renderer';
// import { View } from 'react-native';

// it(`renders correctly Home Screen`, () => {
//     const tree = renderer.create(<View>Snapshot test!</View>).toJSON();
  
//     expect(tree).toMatchSnapshot();
// });
  


// // Mock de los componentes importados
// jest.mock('@/components/TopBar', () => () => <Text>TopBar</Text>);
// jest.mock('@/components/SearchInput', () => () => <Text>SearchInput</Text>);
// jest.mock('@/components/ShowFinancialProducts', () => () => <Text>ShowFinancialProducts</Text>);

// // describe('HomeScreen', () => {
//   test('verifies that TopBar component is rendered', () => {
//     const { getByText } = render(
//       <NavigationContainer>
//         <HomeScreen />
//       </NavigationContainer>
//     );

//     const topBar = getByText('TopBar');
//     expect(topBar).toBeTruthy();
//   });

//   test('verifies that SearchInput component is rendered', () => {
//     const { getByText } = render(
//       <NavigationContainer>
//         <HomeScreen />
//       </NavigationContainer>
//     );

//     const searchInput = getByText('SearchInput');
//     expect(searchInput).toBeTruthy();
//   });

//   test('verifies that ShowFinancialProducts component is rendered', () => {
//     const { getByText } = render(
//       <NavigationContainer>
//         <HomeScreen />
//       </NavigationContainer>
//     );

//     const showFinancialProducts = getByText('ShowFinancialProducts');
//     expect(showFinancialProducts).toBeTruthy();
//   });

//   test('verifies that navigation works when pressing ADD button', () => {
//     const { getByText } = render(
//       <NavigationContainer>
//         <HomeScreen />
//       </NavigationContainer>
//     );

//     const addButton = getByText('ADD');
//     fireEvent.press(addButton);

//     // Aquí puedes verificar que la navegación se haya llamado con el argumento correcto
//     // Si usas mocks, podrías usar expect(navigation.navigate).toHaveBeenCalledWith('screens/AddScreen');
//   });
// // });
