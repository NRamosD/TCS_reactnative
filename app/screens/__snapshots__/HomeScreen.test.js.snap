// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly StatusBar 1`] = `null`;

exports[`renders correctly Topbar 1`] = `
<View
  style={
    {
      "alignContent": "center",
      "backgroundColor": "#fff",
      "borderBottomWidth": 1,
      "borderColor": "#e6e6e6",
      "display": "flex",
      "height": 40,
      "justifyContent": "center",
      "position": "absolute",
      "textAlign": "center",
      "top": 25,
      "width": "100%",
      "zIndex": 10,
    }
  }
>
  <View>
    <Text
      style={
        {
          "textAlign": "center",
        }
      }
    >
      ðŸ’µ BANCO
    </Text>
  </View>
</View>
`;
